src/main/asciidoc/security-refresh/security.adoc

```diff
@@ -0,0 +1,410 @@
+= Jakarta Security
+
+Jakarta Security is the overarching security API in Jakarta EE. 
+
+Due to historical and political reasons, a number of security features are distributed among several other APIs in Jakarta EE. Sometimes they overlap, sometimes such features are only accessible from these other APIs. In this chapter we will focus primarily on explaining Jakarta Security, and will mention when other APIs are needed to accomplish a certain task.
+
+== Overview
+
+Before we look at some practical examples, let's quickly go through some basics. 
+
+Some of the guiding principles in Jakarta Security are:
+
+1. It should work directly out of the box, without requiring any kind of vendor specific configuration or activation.
+2. It leverages Jakarta CDI as much as possible. Most artefacts are CDI beans and many features are done via CDI interceptors.
+3. The difference between framework provided artefacts and custom (user provided) artefacts is minimal or non-existent.
+4. It fully integrates with security features from other Jakarta EE APIs and proprietary (vendor specific) artefacts.
+
+Jakarta Security makes an important distinction between several distinct artefacts that play an important role in the security process:
+
+1. The Authentication Mechanism
+2. The Identity Store
+3. The Permission Store
+
+The first two of these are used in the authentication process:
+
+An _authentication mechanism_ is somewhat like a controller in the well-known MVC pattern; it’s the entity that interacts with the caller (typically a human) via some kind of view to collect credentials and with the model (business logic) to validate these credentials. An authentication mechanism knows about the environment that this caller uses to communicate with the server. An authentication mechanism for HTTP knows about URLs to redirect or forward to, or about response headers to send to the client. It also knows about the data coming back, such as cookies, request headers, and post data. Examples of authentication mechanisms are FORM authentication and BASIC authentication.
+
+An _identity store_ on its turn is more like the model in the MVC pattern. This entity strictly performs a business / data operation where credentials go in, and an identity comes out. The identity contains logic to validate said credentials, and embeds or contacts a database. This "database" contains user names, along with their credentials and (typically) roles. An identity store therefore knows nothing about the environment that this caller uses to communicate with the server; e.g. it doesn't know about HTTP or headers etc.
+Example of identity stores are services that contact SQL/JDBC databases, LDAP servers, files on the file-system, etc.
```

Comment:
Well, the provided identity store in Jakarta Security is SQL/JDBC based, hence why I used that there. Obviously in general it could be SQL and NoSQL indeed.